### PR TITLE
fixes issue #76

### PR DESCRIPTION
**Pull request by RicardoAlmeidaMarques**
Fixes an issue where having the same attachment and target attachment (top-left top-left for example) would cause tether to change the vertical attachment twice when out of bounds, essentially making the attachment remain unchanged.

### CODE REVIEW

**Comment by TrevorBurnham** [comment]
Can you walk me through this change? It sounds like [you're saying](https://github.com/RicardoAlmeidaMarques/tether/blob/566ec647cbfa36bd801cdd780e55319ca6127ea6/src/js/constraint.js#L117):

> When `changeAttachY === 'together'` and `tAttachment.top === 'top'`, check `top + height > bounds[3]` instead of `top < bounds[1]` to decide whether to flip the attachment

That is, when attached to the top of the target, flip if we're *below* the constraint rect instead of flipping when we're *above* the constraint rect. That doesn't seem right to me... can you draw me a diagram?

**Comment by RicardoAlmeidaMarques** [comment]
Hey Tevor! What you said made sense and I am going to remake this PR. Let me just show you what the issue I was resolving is.

https://jsfiddle.net/RicardoMarques/s7xtsz1v/ This is using the current Tether dist code. I set the element and the target attachments to 'top left' and put a series of console.logs between lines 1371 and 1401.
You'll see that even scrolling within that box does not cause any of those console.logs to fire, when what we _want_ to see is the closure with `console.log('2nd')` executing.


https://jsfiddle.net/RicardoMarques/wekwhu42/1/ Here is the same example, with the newest changes.

File: dist/js/tether.js [hunk 1]
```diff
@@ -1364,34 +1364,32 @@ TetherBase.modules.push({
...
+        if (tAttachment.top === 'top') {
+          if (eAttachment.top === 'bottom' && top < bounds[1]) {
             top += targetHeight;
             tAttachment.top = 'bottom';
 
             top += height;
             eAttachment.top = 'top';
-          } else if (eAttachment.top === 'top') {
-            top += targetHeight;
+          } else if (eAttachment.top === 'top' && top + height > bounds[3] && top - (height - targetHeight) >= bounds[1]) {
```
Comment (RicardoAlmeidaMarques) [inline dist/js/tether.js]: If both attachment and target attachment are "top left" for example, it doesn't make sense to check against `bounds[1]`. Instead, what we should check, is if the bottom of the element surpasses `bounds[3]` and if there is room to show the element without going above `bounds[1]`. The opposite goes for attachments on "bottom".

**Comment by TrevorBurnham** [comment]
Yep, this makes sense to me now. Thanks, Ricardo. 👏

**Pull request merged**